centreon/src/Core/Common/Infrastructure/Upload/ZipFileIterator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Common\Infrastructure\Upload;
+
+use Symfony\Component\HttpFoundation\File\File;
+
+class ZipFileIterator implements FileIteratorInterface
+{
+    public const MIME_TYPE = 'application/zip';
+
+    private \ZipArchive $zipArchive;
+
+    private int $filePosition = 0;
+
+    private string $currentFilename = '';
+
+    /**
+     * @param File $file
+     *
+     * @throws \Exception
+     */
+    public function __construct(readonly private File $file)
+    {
+        if ($this->file->getMimeType() !== self::MIME_TYPE) {
+            throw new \Exception(
+                sprintf('Incompatible file type: %s != %s', $this->file->getMimeType(), self::MIME_TYPE)
+            );
+        }
+        $this->zipArchive = new \ZipArchive();
+        $openStatus = $this->zipArchive->open($file->getRealPath());
+        if ($openStatus !== true) {
+            throw new \Exception($this->getErrorMessage($openStatus));
+        }
+    }
+
+    public function count(): int
+    {
+        return $this->zipArchive->count();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function current(): string
+    {
+
+        $currentFilename = $this->zipArchive->getNameIndex($this->filePosition);
+        if ($currentFilename === false) {
+            throw new \Exception();
+        }
+        $this->currentFilename = $currentFilename;
```

Comment:
In the classic use of a `foreach`, it works because `current()` is called before `key()`, but to be more reliable, you should implement the assignement logic of `currentFilename` + `filePosition` in `rewind()` + `next()` OR in `valid()`. cf. [iterator](https://www.php.net/manual/en/class.iterator.php)

Reply:
Yes, even if it won't change anything from a functional point of view.